### PR TITLE
Bug 1797789: [Doc] Update doc for adding "fieldDependency" specDescriptor

### PR DESCRIPTION
**Pull request by tlwu2013**
Add **fieldDependency** specDescriptor:
fielddependency

### CODE REVIEW

**Comment by tlwu2013** [comment]
This the "doc only" PR for adding **fieldDependency** specDescriptor. (cc @alimobrem, @yzhao583 )

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md [hunk 1]
```diff
@@ -676,7 +677,63 @@ This descriptor is created specifically for Prometheus Operator to specify a lis
...
+…
+```
+
+**UI**
+<table style="width:100%">
+  <tr style="vertical-align:top">
+    <td width="50%">CREATION VIEW
+      <img src="img/20-1_fielddependency-new.png" />
+      <br>
+      <small><b>* </b><i> When "Enable Upgrade" (CONTROL FIELD) is TRUE:</i></small>
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md]:
```suggestion
      <small><b>* </b><i> When "Enable Upgrades" (CONTROL FIELD) is TRUE:</i></small>
```

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md [hunk 2]
```diff
@@ -676,7 +677,63 @@ This descriptor is created specifically for Prometheus Operator to specify a lis
 </table>
 
 
-### 20. DEPRECATED Descriptors
+### 20. Field Dependency
+
+**x-descriptors**
+
+This descriptor allows you to specify a field as the dependent of a Control Field and being visible/present only when the current value of the Control Field is equal to the expected value:
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md]:
```suggestion
This descriptor allows you to specify a field as the dependent of a Control Field and shows the field when the current value of the Control Field is equal to the expected value:
```

**Comment by tlwu2013** [comment]
Thanks @spadgett !  I've included your suggestions and fix some minor typos.

**Pull request merged**